File: server/src/player.ts
export default class Player {
  name: string;
  score: number;

  constructor(name: string) {
    this.name = name;
    this.score = 0;
  }

  increaseScore(value: number) {
    this.score += value;
  }

  decreaseScore(value: number) {
    this.score -= value;
  }
}

File: back/player.go
package main

import (
	"github.com/google/uuid"
	"github.com/gorilla/websocket"
)

// TODO: Also add WebRTC signalling data here
type Player struct {
	name   string
	score  int
	conn   *websocket.Conn
	isHost bool
	id     uuid.UUID
}

func createNewPlayer(conn *websocket.Conn) *Player {
	return &Player{
		name:   "",
		score:  0,
		conn:   conn,
		isHost: false,
		id:     uuid.New(),
	}
}
